client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -984,11 +995,25 @@ public void unregisterShuffle(String appId, int shuffleId) {
                   ShuffleServerClientFactory.getInstance()
                       .getShuffleServerClient(clientType, shuffleServerInfo, rssConf);
               RssUnregisterShuffleResponse response = client.unregisterShuffle(request);
-              if (response.getStatusCode() != StatusCode.SUCCESS) {
-                LOG.warn("Failed to unregister shuffle to " + shuffleServerInfo);
+              if (response.getStatusCode() == StatusCode.SUCCESS) {
+                LOG.info("Successfully unregistered shuffle from {}", shuffleServerInfo);
+              } else {
+                LOG.warn("Failed to unregister shuffle from {}", shuffleServerInfo);
               }
             } catch (Exception e) {
-              LOG.warn("Error happened when unregistering to " + shuffleServerInfo, e);
+              if (e.getCause() != null && e.getCause() instanceof InterruptedException) {
+                // InterruptedException indicates timeout, give some advice
```

Comment:
Why the `InterruptedException` indicates the timeout? I think you could directly use the `CompletableFutureExtension` that has been created in uniffle common module.

Reply:
This `InterruptdException` is caused by `ExecutorService.invokeAll` interrupting all threads on (general) timeout. Further, from the `Future` point of view, I would not be able to easily access the `shuffleServerInfo` to generate this error message for the interrupted request.

Reply:
Good to know this. will the CompletableFutureExtension solve this elegantly?

Reply:
I can't see how to integrate `CompletableFutureExtension` here.

Reply:
I could move detecting `ExecutorService.invokeAll` timeouts in `ThreadUtils.executeTasks`: bed40c7b51dcaa35965ed34500fd3c2a1e6097af

Reply:
> I could move detecting `ExecutorService.invokeAll` timeouts into `ThreadUtils.executeTasks`: [8ded662](https://github.com/apache/incubator-uniffle/commit/8ded662f9505733c01de97d1b1949a3e242fba89)

Now it looks good to me. Actually the similar things has beed applied in `CompletableFutureExtension`